packages/liveblocks-react/src/factory.tsx

```diff
@@ -550,6 +551,19 @@ export function createRoomContext<
     );
   }
 
+  function useOthersListener(
+    callback: (event: OthersEvent<TPresence, TUserMeta>) => void
+  ) {
+    const room = useRoom();
+    const savedCallback = useLatest(callback);
+
+    React.useEffect(
+      () =>
+        room.events.others.subscribe((event) => savedCallback.current(event)),
+      [room, savedCallback]
```

Comment:
**Nitpick:** I am guessing we save this callback (as `savedCallback`) so that the effect doesn't rerun when the callback changes. But, I think in this case, it might be okay (not a big deal) to let the effect rerun when the callback changes. Also, I do not expect this callback to change on every render, and if it does become a bother for user that this effect is rerunning multiple times, they could apply this optimization on their own end (i.e., save the callback themselves before passing to the `useOthersListener` hook). 

And, I know it's okay to keep using `useLatest` here but I see the opportunity to use less hooks, which would mean less things to understand for someone new joining or looking at the code.

Reply:
Thanks for the clarification @nvie . It is a difficult balance to make. I think this experimental hook will help solve some of the issues once released and I think you might find it interesting - [useEffectEvent](https://react.dev/learn/separating-events-from-effects)

Reply:
Yeah, I recently read about it! Not sure if I'm a fan of the API yet. I definitely see the problem it solves, but I also think using our `useLatest` construct, and relying on `ref.current` isn't that much worse. At least there are fewer caveats and special cases with this. No new concepts to learn if you already understand callbacks, effects, and refs.

What worries me a bit is how specially the result of `useEffectEvent` is (going to be) treated. It should never be used outside of an effect (i.e. doing `setTimeout(onSomething, 100)` are considered wrong, and linters will have to start catching that).

Also, the `exhaustive-deps` ESLint rule will treat it specially, as if it's a normal callback function you would have to put in the dep array, except… if the variable is the result of a call to `useEffectEvent`, you may not. I think with custom hooks already, but with more special cases like this, `exhaustive-deps` will stop scaling over time.

Comparing their example:

```tsx
function ChatRoom({ roomId, theme }) {
  const onConnected = useEffectEvent(() => {
  //                  ^^^^^^^^^^^^^^ 1️⃣
    showNotification('Connected!', theme);
  });

  useEffect(() => {
    const connection = createConnection(serverUrl, roomId);
    connection.on('connected', () => {
      onConnected();
//    ^^^^^^^^^^^^^ 2️⃣
    });
    connection.connect();
    return () => connection.disconnect();
  }, [roomId]);
  // ...
```

vs

```tsx
function ChatRoom({ roomId, theme }) {
  const onConnected = useLatest(() => {
  //                  ^^^^^^^^^ 1️⃣
    showNotification('Connected!', theme);
  });

  useEffect(() => {
    const connection = createConnection(serverUrl, roomId);
    connection.on('connected', () => {
      onConnected.current();
//    ^^^^^^^^^^^^^^^^^^^ 2️⃣
    });
    connection.connect();
    return () => connection.disconnect();
  }, [roomId, /* (optionally) onConnected */]);
//            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 3️⃣
  // ...
```

To me I don't think we're that far apart. But maybe I'm missing something here, so take it with a grain of salt.